### PR TITLE
[WIP] Emit warning for empty config object

### PR DESCRIPTION
**Pull request by Endlessline**
Newbie here, hopefully this is getting to the right area.

### CODE REVIEW

**Comment by alcuadrado** [comment]
Hey @Endlessline,

thanks for submitting this PR, but I'm a bit confused about it, as an empty config is perfectly valid. Can you explain what was your intention here, please? Thanks!

File: packages/hardhat-core/src/internal/context.ts [hunk 1]
```diff
@@ -78,16 +78,25 @@ export class HardhatContext {
   }
 
   public getFilesLoadedDuringConfig(): string[] {
```
Comment (alcuadrado) [inline packages/hardhat-core/src/internal/context.ts:80]: This is an unrelated concept. What we want to check is the `userConfig` here: https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/src/internal/core/config/config-loading.ts#L66

**Comment by fvictorio** [comment]
@Endlessline the commits in this PR seem completely off. Maybe you did a wrong merge/rebase or something like that?

**Comment by Endlessline** [comment]
Good looking out @fvictorio, I reset the head in my fork and closed this out. I'll start anew so as not to muddy up the history.
